src/compiler/parser.ts

```diff
@@ -542,6 +524,16 @@ namespace ts {
         let TokenConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
         let IdentifierConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
         let SourceFileConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
+        interface Fail extends Node { kind: SyntaxKind.Unknown; }
+        interface FailList extends NodeArray<Node> { pos: -1; }
+        let Fail: Fail;
+        let FailList: FailList;
+        function isFail(x: Node | undefined): x is Fail {
+            return !!x && x.kind === SyntaxKind.Unknown;
+        }
+        function isFailList(x: NodeArray<Node> | undefined): x is FailList {
+            return !!x && x.pos === -1;

```

Comment:
`return FailList !== undefined && x === FailList;` doesn't require the property lookup and verifies that `FailList` has a value.